Mage.Sets/src/mage/sets/lorwyn/EntanglingTrap.java

```diff
@@ -101,13 +101,24 @@ public boolean checkEventType(GameEvent event, Game game) {
 
     @Override
     public boolean checkTrigger(GameEvent event, Game game) {
-        if (event.getPlayerId().equals(getControllerId())) {
-            if (event.getFlag()) { // clash won
-                addEffect(new DontUntapInControllersNextUntapStepTargetEffect("that creature"));
+        //remove effects from previous triggers
+        List<Effect> effects = getEffects();
+        List<Effect> effectsToRemove = new ArrayList<>();
+        for (Effect effect : effects) {
+            if (effect instanceof DontUntapInControllersNextUntapStepTargetEffect) {
+                effectsToRemove.add(effect);
             }
-            return true;
         }
-        return false;
+        for (Effect effect : effectsToRemove) {
+            effects.remove(effect);
+        }
+
+        if (event.getData().equals("controller") && event.getPlayerId().equals(getControllerId())
+                || event.getData().equals("opponent") && event.getTargetId().equals(getControllerId())) {
+            addEffect(new DontUntapInControllersNextUntapStepTargetEffect());
+        }
+
+        return true;

```

Comment:
This code _seems_ incomplete to me. You have checked for the clash event type, but not the result of the clash. You're also doing a check of the event data which contains simple strings? I think this could all be perfectly correct and that the two discrepancies complement one another and if that's the case then okay, but are you sure?

Does this event data for the clash contain these strings and they confirm who wont he clash?